spec/support/matchers/memory_matchers.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'memory_profiler'
+
+RSpec::Matchers.define :allocate_less_than do |bytes|
+  supports_block_expectations
+
+  match do |block|
+    @expected = bytes.to_i
+
+    report = MemoryProfiler.report(ignore_files: /rspec/) do
+      # NOTE(ezekg) reenable GC because profiler disables it but we want
+      #             to measure memory consumption under normal load
```

Comment:
Add note that we don't care about total memory consumed w/o GC but rather total memory footprint w/ GC.